### ISSUE TITLE
 Is it possible to ignore the directory "foo/.a/b/c" but keep the directory "foo/c/"

### ISSUE THREAD

**Issue by pingfury108** [laixintao/prometheus-http-sd#58]
Is it possible to ignore the directory "foo/.a/b/c" but keep the directory "foo/c/"
I tried modifying it, but I don’t know if I can submit pull requests. 

codes: 
  - [ignore directory](https://github.com/pingfury108/prometheus-http-sd/blob/main/prometheus_http_sd/app.py#L100C68-L100C68)
  - [ignore file](https://github.com/pingfury108/prometheus-http-sd/blob/main/prometheus_http_sd/sd.py#L54C60-L54C60)

**Comment by laixintao** [comment]
Hi, sorry I don't quite follow you.

Currently, if the directory like this:

```
.
└── foo
    ├── .a
    │   └── b
    │       └── c
    └── c

```

it should behave exactly like what you said:

- the `.a` and subdirectories are ignored.
- the `c` not.

**Comment by pingfury108** [comment]
My directory looks like this:
```
foo
├── .a
│   └── b
│       └── 3.txt
├── e
│   └── 1.yaml
└── f
    ├── __pycache__
    │   └── f.cpython-312.pyc
    └── f.py
```
This is the startup parameter: `prometheus-http-sd  serve foo`.

Open the homepage like this:
![Snipaste_2023-12-31_15-50-10](https://github.com/laixintao/prometheus-http-sd/assets/27534888/cae3788e-064d-4dd0-8273-50b5fdf3acc1)

Target contains .a and subdirectories, an exception occurred when accessing http://127.0.0.1:8080/targets/.a, log: 
```
[2023-12-31 15:50:23,242] {app.py:1414} ERROR - Exception on /targets/.a [GET]
Traceback (most recent call last):
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/prometheus_http_sd/app.py", line 80, in get_targets
    targets = generate(config.root_dir, rest_path, **request.args)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/prometheus_http_sd/sd.py", line 104, in generate
    target_list = run_generator(generator, **extra_args)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pingfury/Library/Application Support/pipx/venvs/prometheus-http-sd/lib/python3.12/site-packages/prometheus_http_sd/sd.py", line 121, in run_generator
    raise Exception(f"Unknown File Type: {generator_path}")
Exception: Unknown File Type: foo/.a/b/3.txt
`
``

I want:
- Ignore .a and subdirectories
- .a and subdirectories are not in the homepage target list

**Comment by pingfury108** [comment]
Sorry, I closed issues by mistake.

**Comment by laixintao** [comment]
Thanks, this was a bug: seems only file name started with `.` being ignored. now should be fixed.

**Comment by laixintao** [comment]
released in v1.3.7
